### DISCUSSION TITLE
É possível organizar os fontes em diretórios separados dentro da pasta src ou os fontes precisam necessariamente estar na src?

### DISCUSSION THREAD

**Discussion by mateus-sf** [HashLoad/boss#118] [Q&A] [answered]
Motivo da pergunta: Preciso trabalhar compartilhando com os outros programadores da equipe apenas os .dcu ao invés dos .pas. O problema disso é que preciso gerar o .dcu para cada plataforma (Android32, Android64, Win32, ...). Consequentemente, vou ter vários .dcu com o mesmo nome e função, a única diferença é que cada um atende uma plataforma diferente. Minha solução foi organizar a pasta src com divisões por plataforma (src/Android32, src/Android64, src/Win32...) e os .dcu seriam alocados cada um em sua respectiva pasta. Minha dúvida é se o boss trata a mainsrc recursivamente e adiciona as subpastas ao search path.

**Comment by andriwsluna** [comment, marked as answer]
Sim, ele percorre todas as pastas e, se a pasta tiver algum arquivo com extensão que atenda à regex 
`RegexArtifacts = "(.*.inc$|.*.pas$|.*.dfm$|.*.fmx$|.*.dcu$|.*.bpl$|.*.dcp$)"`
, ele adiciona ao path.
